pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -547,6 +548,11 @@ func TestDesiredRouterDeployment(t *testing.T) {
 
 	checkDeploymentHasEnvSorted(t, deployment)
 
+	// verify the default ports are used
+	checkContainerPort(t, deployment, "http", 80)
+	checkContainerPort(t, deployment, "https", 443)
+	checkContainerPort(t, deployment, "metrics", 1936)
+
```

Comment:
Might as well check the corresponding environment variables too.

Reply:
The environment variables for the HTTP and HTTPS have defaults and currently those defaults are used. I've added a check for the stats port environment variable because that's always set regardless of the strategy. Should I update the code to always set those environment variables as well?

Reply:
Ah, I see what you're saying—the environment variables are only set if the endpoint publishing strategy is "HostNetwork".  Never mind then.